Core/AVIDump.cpp

```diff
@@ -0,0 +1,360 @@
+	const u8 *buffer = ConvertBufferTo888RGB(buf, flipbuffer, w, h);
+	s_src_frame->data[0] = const_cast<u8*>(buffer);
+	s_src_frame->linesize[0] = w * 3;
+	s_src_frame->format = AV_PIX_FMT_RGB24;
+	s_src_frame->width = s_width;
+	s_src_frame->height = s_height;
+
+	// Convert image from BGR24 to desired pixel format, and scale to initial
+	// width and height
+	if ((s_sws_context =
+           sws_getCachedContext(s_sws_context, w, h, AV_PIX_FMT_RGB24, s_width, s_height,
+                                s_stream->codec->pix_fmt, SWS_BICUBIC, nullptr, nullptr, nullptr)))
+	{
+		sws_scale(s_sws_context, s_src_frame->data, s_src_frame->linesize, 0, h,
+              s_scaled_frame->data, s_scaled_frame->linesize);
+	}
+
+	s_scaled_frame->format = s_stream->codec->pix_fmt;
+	s_scaled_frame->width = s_width;
+	s_scaled_frame->height = s_height;
+
+	// Encode and write the image.
+	AVPacket pkt;
+	PreparePacket(&pkt);
+	int got_packet;
+	int error = avcodec_encode_video2(s_stream->codec, &pkt, s_scaled_frame, &got_packet);
+	while (!error && got_packet)
+	{
+		// Write the compressed frame in the media file.
+		if (pkt.pts != (s64)AV_NOPTS_VALUE)
+		{
+			pkt.pts = av_rescale_q(pkt.pts, s_stream->codec->time_base, s_stream->time_base);
+		}
+		if (pkt.dts != (s64)AV_NOPTS_VALUE)
+		{
+			pkt.dts = av_rescale_q(pkt.dts, s_stream->codec->time_base, s_stream->time_base);
+		}
+#if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(56, 60, 100)
+		if (s_stream->codec->coded_frame->key_frame)
+			pkt.flags |= AV_PKT_FLAG_KEY;
+#endif
+		pkt.stream_index = s_stream->index;
+		av_interleaved_write_frame(s_format_context, &pkt);
+
+		// Handle delayed frames.
+		PreparePacket(&pkt);
+		error = avcodec_encode_video2(s_stream->codec, &pkt, nullptr, &got_packet);
+	}
+	if (error)
+		ERROR_LOG(G3D, "Error while encoding video: %d", error);
+}
+
+void AVIDump::Stop()
+{
+	av_write_trailer(s_format_context);
+	CloseFile();
+	s_file_index = 0;
+	NOTICE_LOG(G3D, "Stopping frame dump");
+}
+
+void AVIDump::CloseFile()
+{
+	if (s_stream)
+	{
+		if (s_stream->codec)
+		{
+#if LIBAVCODEC_VERSION_MAJOR < 55
+			avcodec_default_release_buffer(s_stream->codec, s_src_frame);
+#endif
+			avcodec_close(s_stream->codec);
+		}
+		av_freep(&s_stream);
+	}
+
+	av_frame_free(&s_src_frame);
+	av_frame_free(&s_scaled_frame);
+
+	if (s_format_context)
+	{
+		if (s_format_context->pb)
+			avio_close(s_format_context->pb);
+		av_freep(&s_format_context);
+	}
+
+	if (s_sws_context)
+	{
+		sws_freeContext(s_sws_context);
+		s_sws_context = nullptr;
+	}
+}
+
+void AVIDump::CheckResolution(int width, int height)
+{
+	// We check here to see if the requested width and height have changed since the last frame which
+	// was dumped, then create a new file accordingly. However, is it possible for the height
+	// (possibly width as well, but no examples known) to have a value of zero. This can occur as the
+	// VI is able to be set to a zero value for height/width to disable output. If this is the case,
```

Comment:
This seems like a very Dolphiny comment.  In PPSSPP, the size can change at any time - but the displayed size is ALWAYS the top left 480x272.  So it may make more sense to just always crop.

Also, there's a "display" framebuffer and a "render" framebuffer.  I'm not sure which one this code is grabbing... the GE debugger uses the "render" one so that might be what's happening here too.  We can probably add a flag.  Just a warning in case there are issues (could cause black screens in some games.)

-[Unknown]